### PR TITLE
A few fixes around libp2p-xtra

### PR DESCRIPTION
**Pull request by klochowicz**
Additions and fixes discovered during integration

### CODE REVIEW

File: libp2p-xtra/examples/dialer.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+struct Opts {
+    #[clap(long)]
+    multiaddr: Multiaddr,
+
+    #[clap(long, default_value = "ExampleDialer")]
+    name: String,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
```
Comment (thomaseizinger) [inline libp2p-xtra/examples/dialer.rs]: Perhaps we should call the file `hello_world_dialer`.

File: libp2p-xtra/src/endpoint.rs [hunk 2]
```diff
@@ -381,30 +382,48 @@ impl Endpoint {
         let this = ctx.address().expect("we are alive");
         let listen_address = msg.0.clone();
 
-        self.listen_addresses.insert(listen_address.clone()); // FIXME: This address could be a "catch-all" like "0.0.0.0" which actually results in
-                                                              // listening on multiple interfaces.
```
Comment (thomaseizinger) [inline libp2p-xtra/src/endpoint.rs:392]: If possible, can we keep this one here and remove the line in the last commit with the rationale that we want to wait until the call to `listen_on` succeeds before we add it?

**Review by thomaseizinger — APPROVED** [review]

File: libp2p-xtra/src/endpoint.rs (hunk 2 above)
Comment (klochowicz) [reply libp2p-xtra/src/endpoint.rs:392]: I'm afraid I don't follow. are you talking about the  the comment or the whole 2 lines? and remove the whole solution from last commit?

Comment (thomaseizinger) [reply libp2p-xtra/src/endpoint.rs:392]: I think this commit should not touch these two lines and instead we can move the insert from here to what we have in the last commit!

**Pull request merged**